Idle get_amounts fails when gauge_address is None

It is not clear to me why this functions is using the gauge decimals to scale both the bb_balance and the aa_balance. It seems to be a bug to me. If all the decimals are the same value then it is better to use the aa_contract decimals, that is always present right? If they may be different then use each specific decimal for each contract.

This issue makes get_underlying_all to always fail!

```python
underlying_address = '0x956F47F50A910163D8BF957Cf5846D573E7f87CA', cdo_address = '0x77648a2661687ef3b05214d824503f6717311596'
aa_address = '0x9cE3a740Df498646939BcBb213A66BBFa1440af6', bb_address = '0x2490D810BF6429264397Ba721A488b0C439aA745', gauge_address = None
wallet = '0x849D52316331967b6fF1198e5E32A0eB168D039d', block = 16836190, blockchain = 'ethereum', web3 = <web3.main.Web3 object at 0x7f739814d9f0>, decimals = True

    def get_amounts(underlying_address: str, cdo_address: str, aa_address: str, bb_address: str, gauge_address: str,
                    wallet: str, block: Union[int, str], blockchain: str, web3=None, decimals: bool = True) -> list:
        balances = []
    
        if web3 is None:
            web3 = get_node(blockchain, block=block)
    
        wallet = web3.to_checksum_address(wallet)
        cdo_contract = get_contract(cdo_address, blockchain, web3=web3, abi=ABI_CDO_IDLE, block=block)
        aa_contract = get_contract(aa_address, blockchain, web3=web3, abi=ABI_TOKEN_SIMPLIFIED, block=block)
        bb_contract = get_contract(bb_address, blockchain, web3=web3, abi=ABI_TOKEN_SIMPLIFIED, block=block)
        if gauge_address:
            gauge_contract = get_contract(gauge_address, blockchain, web3=web3, abi=ABI_GAUGE, block=block)
            gauge_balance = gauge_contract.functions.balanceOf(wallet).call(block_identifier=block) * (
                        cdo_contract.functions.virtualPrice(aa_address).call() / Decimal(10 ** 18))
        else:
            gauge_balance = 0
        aa_balance = aa_contract.functions.balanceOf(wallet).call(block_identifier=block) * (
                    cdo_contract.functions.virtualPrice(aa_address).call() / Decimal(10 ** 18))
        bb_balance = bb_contract.functions.balanceOf(wallet).call(block_identifier=block) * (
                    cdo_contract.functions.virtualPrice(bb_address).call() / Decimal(10 ** 18))

        decimals_underlying = gauge_contract.functions.decimals().call()    
E       UnboundLocalError: local variable 'gauge_contract' referenced before assignment
```

--- a/defi_protocols/Idle.py
+++ b/defi_protocols/Idle.py
@@ -55,7 +55,6 @@ ABI_GAUGE: str = '[{"stateMutability":"view","type":"function","name":"decimals"
 
 ABI_CDO_PROXY: str = '[{"anonymous":false,"inputs":[{"indexed":false,"internalType":"address","name":"proxy","type":"address"}],"name":"CDODeployed","type":"event"},{"inputs":[{"internalType":"address","name":"implementation","type":"address"},{"internalType":"address","name":"admin","type":"address"},{"internalType":"bytes","name":"data","type":"bytes"}],"name":"deployCDO","outputs":[],"stateMutability":"nonpayable","type":"function"}]'
 
-
 # function for getting all addresses you need to get underlying
 def get_addresses(block: Union[int, str], blockchain: str, web3=None, execution: int = 1, index: int = 0,
                   decimals: bool = True) -> list:

--- a/defi_protocols/Idle.py
+++ b/defi_protocols/Idle.py
@@ -168,9 +168,6 @@ def get_all_rewards(wallet: str, gauge_address: str, block: Union[int, str], blo
         return get_all_rewards(wallet, gauge_address, block, blockchain, web3=None, index=index + 1,
                                execution=execution, decimals=decimals)
 
-    # function for getting the amounts of certain tranche
-
-
 def get_amounts(underlying_address: str, cdo_address: str, aa_address: str, bb_address: str, gauge_address: str,
                 wallet: str, block: Union[int, str], blockchain: str, web3=None, execution: int = 1, index: int = 0,
                 decimals: bool = True) -> list:

--- a/defi_protocols/Idle.py
+++ b/defi_protocols/Idle.py
@@ -197,7 +197,7 @@ def get_amounts(underlying_address: str, cdo_address: str, aa_address: str, bb_a
                     cdo_contract.functions.virtualPrice(aa_address).call() / Decimal(10 ** 18))
         bb_balance = bb_contract.functions.balanceOf(wallet).call(block_identifier=block) * (
                     cdo_contract.functions.virtualPrice(bb_address).call() / Decimal(10 ** 18))
-        decimals_underlying = gauge_contract.functions.decimals().call()
+        decimals_underlying = aa_contract.functions.decimals().call()
 
         if decimals is True:
             [balances.append([underlying_address, float(i / Decimal(10 ** decimals_underlying))]) for i in

